Error during installation: 'warning: possible EventEmitter memory leak detected. 11 error listeners ...'

# Environment
OS: Windows 10
Atom -v:
```
Atom    : 1.12.6
Electron: 1.3.9
Chrome  : 52.0.2743.82
Node    : 6.5.0
```

# Problem

Trying to install this linter through: `apm install linter-xo`
This is what I get: 
```
> buffer-offset-index@0.0.1 install C:\Users\Giacomo\AppData\Local\Temp\apm-install-dir-116114-5988-eh894t\node_modules\
linter-xo\node_modules\buffer-offset-index
> node-gyp rebuild


C:\Users\Giacomo\AppData\Local\Temp\apm-install-dir-116114-5988-eh894t\node_modules\linter-xo\node_modules\buffer-offset
-index>if not defined npm_config_node_gyp (node "C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_m
odules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "C:\Users\Giacomo\AppDat
a\Local\atom\app-1.12.6\resources\app\apm\bin\\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )
C:\Users\Giacomo\AppData\Local\Temp\apm-install-dir-116114-5988-eh894t
`-- (empty)


(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to i
ncrease limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\npm-registry-c
lient\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\request\
request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to i
ncrease limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\npm-registry-c
lient\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\request\
request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to i
ncrease limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\npm-registry-c
lient\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\request\
request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to i
ncrease limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\npm-registry-c
lient\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\request\
request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to i
ncrease limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\npm-registry-c
lient\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\request\
request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to i
ncrease limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\npm-registry-c
lient\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\request\
request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to i
ncrease limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\npm-registry-c
lient\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\request\
request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to i
ncrease limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\npm-registry-c
lient\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\request\
request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to i
ncrease limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\npm-registry-c
lient\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\request\
request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to i
ncrease limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\npm-registry-c
lient\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\request\
request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to i
ncrease limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\npm-registry-c
lient\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\request\
request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to i
ncrease limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\npm-registry-c
lient\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\request\
request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to i
ncrease limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\npm-registry-c
lient\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\request\
request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to i
ncrease limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\npm-registry-c
lient\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\request\
request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
gyp ERR! configure error
gyp ERR! stack Error: Python executable "C:\Users\Giacomo\AppData\Local\Programs\Python\Python35\python.EXE" is v3.5.1,
which is not supported by gyp.
gyp ERR! stack You can pass the --python switch to point to Python >= v2.5.0 & < 3.0.0.
gyp ERR! stack     at failPythonVersion (C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\n
ode-gyp\lib\configure.js:454:14)
gyp ERR! stack     at C:\Users\Giacomo\AppData\Local\atom\app-1.12.6\resources\app\apm\node_modules\node-gyp\lib\configu
re.js:443:9
gyp ERR! stack     at ChildProcess.exithandler (child_process.js:204:7)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at maybeClose (internal/child_process.js:827:16)
gyp ERR! stack     at Socket.<anonymous> (internal/child_process.js:319:11)
gyp ERR! stack     at emitOne (events.js:77:13)
gyp ERR! stack     at Socket.emit (events.js:169:7)
gyp ERR! stack     at Pipe._onclose (net.js:477:12)
gyp ERR! System Windows_NT 10.0.14931
gyp ERR! command "C:\\Users\\Giacomo\\AppData\\Local\\atom\\app-1.12.6\\resources\\app\\apm\\bin\\node.exe" "C:\\Users\\
Giacomo\\AppData\\Local\\atom\\app-1.12.6\\resources\\app\\apm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\Giacomo\AppData\Local\Temp\apm-install-dir-116114-5988-eh894t\node_modules\linter-xo\node_modules\
buffer-offset-index
gyp ERR! node -v v4.4.5
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok
npm WARN enoent ENOENT: no such file or directory, open 'C:\Users\Giacomo\AppData\Local\Temp\apm-install-dir-116114-5988
-eh894t\package.json'
npm WARN apm-install-dir-116114-5988-eh894t No description
npm WARN apm-install-dir-116114-5988-eh894t No repository field.
npm WARN apm-install-dir-116114-5988-eh894t No README data
npm WARN apm-install-dir-116114-5988-eh894t No license field.
npm ERR! Windows_NT 10.0.14931
npm ERR! argv "C:\\Users\\Giacomo\\AppData\\Local\\atom\\app-1.12.6\\resources\\app\\apm\\bin\\node.exe" "C:\\Users\\Gia
como\\AppData\\Local\\atom\\app-1.12.6\\resources\\app\\apm\\node_modules\\npm\\bin\\npm-cli.js" "--globalconfig" "C:\\U
sers\\Giacomo\\.atom\\.apm\\.apmrc" "--userconfig" "C:\\Users\\Giacomo\\.atom\\.apmrc" "install" "C:\\Users\\Giacomo\\Ap
pData\\Local\\Temp\\d-116114-5988-1lhu3xi\\package.tgz" "--runtime=electron" "--target=1.3.9" "--arch=ia32" "--global-st
yle"
npm ERR! node v4.4.5
npm ERR! npm  v3.10.5
npm ERR! code ELIFECYCLE

npm ERR! buffer-offset-index@0.0.1 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the buffer-offset-index@0.0.1 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the buffer-offset-index package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs buffer-offset-index
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls buffer-offset-index
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\Giacomo\AppData\Local\Temp\apm-install-dir-116114-5988-eh894t\npm-debug.log
npm ERR! code 1
```

PS = I was undecided if putting this huge error would have been a good thing. If it's too long I'll edit the issue.

--- a/lib/format.js
+++ b/lib/format.js
@@ -1,6 +1,5 @@
 /** @babel */
-import {Range} from 'text-buffer';
-// atom-linter prevents unit testing by depending on 'atom' directly
+import {Range} from 'atom';
 import {rangeFromLineNumber} from 'atom-linter';
 import ruleURI from 'eslint-rule-documentation';
 

--- a/lib/format.test.js
+++ b/lib/format.test.js
@@ -1,0 +1,39 @@
+import test from 'ava';
+import proxyquire from 'proxyquire';
+import {Range} from 'text-buffer';
+import MockEditor from '../fixtures/mock-editor';
+
+const stubs = {
+	atom: {
+		Range,
+		'@noCallThru': true,
+		'@global': true
+	}
+};
+
+const editor = new MockEditor({path: '__fake'});
+
+const format = proxyquire('./format', stubs).default;
+
+test('for an empty report', t => {
+	const actual = format(editor)();
+	const expected = [];
+	t.deepEqual(actual, expected, 'it should return an empty array');
+});
+
+test('for a report with an error message', t => {
+	const input = {
+		results: [{
+			messages: [
+				{
+					message: 'message',
+					severity: 2
+				}
+			]
+		}]
+	};
+
+	const [actual] = format(editor)(input);
+	t.is(actual.type, 'Error', 'it should return an Error');
+	t.is(actual.html, '<span>message</span>', 'it should return appropriate html');
+});

--- a/package.json
+++ b/package.json
@@ -36,7 +36,6 @@
     "p-props": "^1.0.0",
     "pkg-dir": "^1.0.0",
     "resolve-from": "^2.0.0",
-    "text-buffer": "^9.4.3",
     "xo": "^0.17.0"
   },
   "devDependencies": {
@@ -46,6 +45,8 @@
     "delay": "^1.3.1",
     "electron-rebuild": "^1.4.0",
     "p-timeout": "^1.0.0",
+    "proxyquire": "^1.7.10",
+    "text-buffer": "^9.4.3",
     "tmp": "0.0.31"
   },
   "package-deps": [

